### PR TITLE
Shouting tweaks and new command to shout at will.

### PR DESCRIPTION
**Pull request by Maykeye**
Common suggestion thread on forum mentions "Player making noise to attract or distract zombies" which is not yet implemented. This PR adds command to shout and reworks existing shouting for consistency.
- Added command to shout at will (default keybinding - '0').
- All screaming done by player (through command, SHOUT{N} mutation or
  due schizophrenia) now takes into account mouth encumbrance so
  wearing a lot of gear on mouth will prevent ability to shout loudly.
- Shouting underwater reduces oxygen.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Shouting underwater doesn't really make sense. Especially not without a big penalty.

I'd integrate the shouting messages into the shout function. Shouter mutations shouldn't really help you taunt.

Having shout bound by default under a key with no shift and near numpad sounds slightly dangerous. It could be integrated into 'C'hat menu (as always-available extra option).

Or it could be a separate menu, with options like "shout", "scream" (louder shout) etc. with the louder shouts being the ones available through mutations.
Later it could be extended to have commands to NPCs, like "come back", that would work without having to chat (and thus see and hear the NPC).

**Comment by Barhandar** [comment]
>Or it could be a separate menu, with options like "shout", "scream" (louder shout) etc. with the louder shouts being the ones available through mutations.
>Later it could be extended to have commands to NPCs, like "come back", that would work without having to chat (and thus see and hear the NPC).

I would rather like to see this instead of it occupying an easy-to-misclick button by default.

Also, OP? "Unassigned by default" is an option.

File: src/player.cpp [hunk 1]
```diff
@@ -7804,7 +7839,7 @@ void player::suffer()
                     break;
                 case 9:
                     add_msg(m_bad, _("You have the sudden urge to SCREAM!"));
-                    sounds::sound( pos(), 10 + 2 * str_cur, "AHHHHHHH!");
+                    shout("AHHHHHHH!");
```
Comment (BevapDin) [inline src/player.cpp]: Not your bug, but this should be translated (via `_()`), as are all the other screams.

**Comment by Maykeye** [comment]
@Coolthulhu 
>  It could be integrated into 'C'hat menu (as always-available extra option).
> Or it could be a separate menu, with options like "shout", "scream" (louder shout)

I thought that these mutation make character more accustomed to screaming overall.  

OK, I'll add it as `C`hat submenu with shout which works like SHOUT  (10+2str)  and scream which works as SHOUT2 (15+3str). Along the way it'll fix an issue that now if you are deaf you can't `S`ize up stats (option from chat menu).

@Barhandar 
> Also, OP? "Unassigned by default" is an option.

I really don't like idea that non-debug commands should be enabled by tweaking configuration. 

> And now I want a Wild Wasteland-type trait/enabler mod that makes this cause explosion on player's tile with some low chance.

As far as damaging goes, I originally thought about causing deafness for a characters if they are screaming inside SHELL2  (as sound bounces in the shell) but decided that it has too little of use. 

Also thought about causing some sort of confusion/stun as with flashbang, but decided then it should be done by any loud noise, not just shouting

**Comment by Coolthulhu** [comment]
> OK, I'll add it as Chat submenu with shout which works like SHOUT (10+2str) and scream which works as SHOUT2 (15+3str).

I meant, mutations should give a boost, but they shouldn't boost speech ("come get some"), only shrieking.
I see it like this: regular human can only shout (noise 10 + str), shout mutation either adds a louder shout (the one with 2*str) or boosts the existing one, shout2 adds a shriek and should3 a howl.

Don't make it a submenu, though. Either add the entries to the menu directly or split it into a totally separate menu, but don't make it so you have to go through 2 separate menus. 2 separate menus is OK for debug and configuration, but not command invocation.

**Comment by Coolthulhu** [comment]
Oh and also: once shouting gets a separate menu, shout mutations should no longer be activatable through mutation menu.

**Comment by Maykeye** [comment]
After thinking I went with "loudest yelling/shouting/howling available": 
giving an option on how loud to shout is not too meaningful unless volume difference is about 100  or more: if I want to attract zombies attention with shouting, I doubt I can easily tell difference between volume 20 and 30 or even 40 to choose if I should pick option yell or shout.

File: src/player.cpp [hunk 2]
```diff
@@ -4263,6 +4263,59 @@ void player::pause()
...
+void player::shout( std::string msg )
+{
+    int base = 10;
+    int shout_multiplier = 2;
+
+    // Mutations make shouting louder, they also define the defualt message
+    if ( has_trait("SHOUT2") ) {
+        base = 15;
+        shout_multiplier = 3;
+        if ( msg == "" ) {
```
Comment (Coolthulhu) [inline src/player.cpp]: Should be `msg.empty()`.

File: src/npctalk.cpp [hunk 3]
```diff
@@ -1533,6 +1542,14 @@ std::string dialogue::dynamic_line( const std::string &topic ) const
     } else if( topic == "TALK_DEMAND_LEAVE" ) {
         return _("Now get out of here, before I kill you.");
 
+    } else if( topic == "TALK_SHOUT" ) {
+        alpha->shout();
```
Comment (Coolthulhu) [inline src/npctalk.cpp]: This doesn't consume moves, while NPC-less shouting does.
Shouting should probably be free, it doesn't take much time IRL.

**Pull request merged**